### ISSUE TITLE
The warning 204 when assigning in "do{} while();" condition scope

### ISSUE THREAD

**Issue by KarpovVyacheslav** [pawn-lang/compiler#545]
### Issue description:

The following code results in a warning 204, although it should not:

```pawn
stock func() 
    return 1;

main()
{
    new i = 0;
    do
    {
         printf("%d", i);
    } while((i = func()));//symbol is assigned a value that is never used: "i"
}
```

### Minimal complete verifiable example (MCVE):

See above.

### Workspace Information:

* Compiler version: 3.10.10
* Command line arguments provided (or sampctl version):
* Operating System:

**Comment by Daniel-Cortez** [comment]
> Compiler version: 3.10.10

It's not in 3.10.10, it's in the development version (the `dev` branch).

Anyway, yes, the problem persists in `dev`, but it's already fixed in #533.

**Comment by Daniel-Cortez** [comment]
Fixed in #533.
@pawn-lang Close?
